sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/RowSetUtils.scala

```diff
@@ -135,10 +135,8 @@ object RowSetUtils {
 
       case _ =>
         var i = 0
-        val rowSize = rows.length
         val values = new java.util.ArrayList[String](rowSize)
-        while (i < rowSize) {
-          val row = rows(i)
+        rows.foreach { row =>
```

Comment:
Hmmm generally `while` is faster than `foreach` (https://github.com/databricks/scala-style-guide?tab=readme-ov-file#traversal-and-zipwithindex)

Reply:
It seems the `Array`'s performance is much better than `Seq`.
```
toColumnBasedSet time: 147, toRowBasedSet time: 78
```